nil/cmd/nild/devnet.go

```diff
@@ -125,10 +127,17 @@ func (devnet devnet) generateZeroState(nShards uint32, servers []server) (*execu
 			})
 		}
 	}
+	mainPrivateKey, mainPublicKey, err := nilcrypto.GenerateKeyPair()
+	if err != nil {
+		return nil, err
+	}
+	mainKeys := execution.MainKeys{MainPrivateKey: "0x" + nilcrypto.PrivateKeyToEthereumFormat(mainPrivateKey), MainPublicKey: hexutil.Encode(mainPublicKey)}
+
 	return &execution.ZeroStateConfig{
 		ConfigParams: execution.ConfigParams{
 			Validators: config.ParamValidators{Validators: validators},
 		},
+		MainKeys: mainKeys,
```

Comment:
I'm a bit confused here. I'd suggest to generate yaml config file and share it between all instances.
You don't need to pass `MainPrivateKey` somewhere. All you need for zerostate is only public key.
You can just dump private key into credentials directory and reuse it if it exists.

Reply:
yes, you are right. fixed it and create keys file.